core/src/main/kotlin/com/karumi/kotlinsnapshot/core/SerializationModule.kt

```diff
@@ -0,0 +1,65 @@
+package com.karumi.kotlinsnapshot.core
+
+import kotlin.reflect.KProperty1
+import kotlin.reflect.KVisibility
+import kotlin.reflect.full.memberProperties
+
+interface SerializationModule<in A> {
+
+    fun serialize(value: A?): String
+}
+
+class KotlinSerialization : SerializationModule<Any> {
+
+    override fun serialize(value: Any?): String = when {
+        value == null -> "null"
+        isPrimitive(value) -> value.toString()
+        value is CharSequence -> value.toString()
+        value::class.java.isEnum -> value.toString()
+        value is Iterable<*> -> iterableToString(value)
+        value is Array<*> -> iterableToString(value.toList())
+        value is Map<*, *> -> mapToString(value)
+        value is Pair<*, *> -> pairToString(value)
+        else -> toString(value)
+    }
+
+    private fun isPrimitive(value: Any) =
+        value::class.javaPrimitiveType?.isPrimitive ?: false
+
+    private fun toString(value: Any): String {
+        val anyClass = value::class
+        val className = anyClass.simpleName
+        val fieldValueList = anyClass.memberProperties.joinToString { field ->
+            getFieldValuePair(value, field)
+        }
+        val classBody = addParenthesesIfNotEmpty(fieldValueList)
+        return "$className$classBody"
+    }
+
+    private fun iterableToString(list: Iterable<*>): String =
+        list.joinToString(", ", "[", "]") {
+            serialize(it)
+        }
+
+    private fun mapToString(map: Map<*, *>): String = map.mapValues {
```

Comment:
Same sorting comment, we should be agnostic of the inner map implementation if possible.

Reply:
yes, I'm going to test to know if it is a problem

Reply:
As we discussed, you can try using different `map` implementations (e.g. `LinkedHashMap` and `HashMap`) and see if items are returned in different order (I bet they do).